src/EditorFeatures/CSharp/Formatting/Indentation/SmartTokenFormatter.cs

```diff
@@ -48,15 +48,25 @@ internal class SmartTokenFormatter : ISmartTokenFormatter
 
             var smartTokenformattingRules = _formattingRules;
             var common = startToken.GetCommonRoot(endToken);
-            if (common.ContainsDiagnostics)
+
+            // if there are errors, do not touch lines
+            // Exception: In the case of try-catch-finally block, a try block without a catch/finally block is considered incomplete
+            //            but we would like to apply line operation in a completed try block even if there is no catch/finally block
+            if (common.ContainsDiagnostics && !CloseBraceOfTryBlock(endToken))
```

Comment:
what if the diagnostic was for something else and not for the absence of catch block?